test_cases/reentracy/simple_dao.yaml

```diff
@@ -0,0 +1,7 @@
+description: "Reentrancy (state change after external call)"
+issues:
+- id: "SWC-107"
+  count: 1
+  location:
+  - bytecode_offset: 658
+    line_number: 11
```

Comment:
line_number:15?

Reply:
Actually Mythril reports line number 16, which is the state change *after* the message call:

```
==== State change after external call ====
Type: Warning
Contract: SimpleDAO
Function name: withdraw(uint256)
PC address: 658
The contract account state is changed after an external call. Consider that the called contract could re-enter the function before this state change takes place. This can lead to business logic vulnerabilities.
--------------------
In file: simple_dao.sol:16

credit[msg.sender]-=amount
```

15 and 16 are both possible locations to report the issue. Do we support line ranges?

Reply:
(I'd actually prefer keeping "line" an integer so parsing the config file remains easy).

Reply:
Some things, I'd note. 

First, "line" is a _already_ kind of range and a kind of fuzzing. 
Seocnd, what you are talking about is "or"ing as distinct from a contiguous range, which is different. 

As for keeping the parsing simple, I come back to Einstein's edict: simple as possible but no simpler. Here the problem is not as simple as you make it out to be. Already we are adjusting this to Mythril's current behavior. If another tool writer deicdes differently, by not allowing some sort of flexibility in specification even if it is not initially implemented or used -- either by allowing "or", or some sort of fuzzing --  we are setting ourselves up for more difficultly in inviiting pariticpation

As for simplicity of parsing, I disagree. 

Here is how you could do this in a way that fits pefectly within YAML:

```yaml
position: 
    line: 5

positon:
   line-range: [5, 6]

position:
    line-or: [5, 6]
 ```

This allows the possibility staring out with say line 5, then futzing it to line 6 because we feel that whatever Mythril does has to be the right answer for now. So there's no more work done now than we are currently doing. 

This doesn't preclude proceding as it appears is being done now: sitcking ot a simple line number. But it leaves _open_ the possiblity that sometime in the future of chaning to  a "line-or".

Reply:
While it's good to keep it as simple as possible, there are variants that will need valid alternative locations. How about allowing alternative locations for both `bytecode_offset` and `line_number `? I feel the notion of a range is too fuzzy.

```
description: "Reentrancy (state change after external call)"
 issues:
 - id: "SWC-107"
   count: 1
   location:
   - bytecode_offset: 658
     line_number: 11,12
```

Reply:
Ulttimately I'm okay with whatever others want to do as long as we don't paint ourselves in a corner. 

A range need not be fuzzy. Consider [eth_tx_oder_dependence_2](https://smartcontractsecurity.github.io/Suhabe/index.html#eth_tx_order_dependence_minimal)

Saying that the range is whatever column positions occur in `owner.transfer(reward)` is valid. If this were written as: 

```solidity
owner.
    transfer(reward);
```

Giving a line range for the two lines would also be valid. I know it has been proposed that you give the first line number and leave vague where the ending is. This is a kind of fuzzing like line is. 

So in sum, we currently allow some sort of position fuzzing when a line number is specified. For byteocde offset there is no fuzzing allowed.

 If I understand this correctly, we don't allow for alternate locations to be allowed as a result of a disagreement as to where the problem is, except to give up totally on indicating any sort of location at all. 

If that's what folks want to do, okay. I'd keep in the back of my mind that at some point we may want to change it.

Reply:
I assume then line-number would always be a list:

```
line_numbers: [15,16]
```

or 

```
line_numbers[15]
```
?

Reply:
As we're already discussing this, I'd also rename the `location` field to `locations` because it's a list?

Reply:
How about this? `bytecode_offsets` and `line_numbers` is a list of locations. This would allow the notion of a range e.g. [1,2,3] as well while not getting to complicated with different formats.

```
   locations:
   - bytecode_offsets: [658,659]
     line_numbers: [15,16]
```